packages/nx/bin/run-executor-worker.ts

```diff
@@ -0,0 +1,142 @@
+import { run } from '../src/command-line/run';
+import { Task } from '../src/config/task-graph';
+import { appendFileSync, openSync, writeFileSync, closeSync } from 'fs';
+import { addCommandPrefixIfNeeded } from '../src/utils/add-command-prefix';
+
+setUpOutputWatching();
+
+process.env.NX_CLI_SET = 'true';
+
+interface ExecuteTaskOptions {
+  workspaceRoot: string;
+  outputPath?: string;
+  streamOutput?: boolean;
+  captureStderr?: boolean;
+  onStdout?: (chunk: string) => void;
+  onStderr?: (chunk: string) => void;
+}
+
+let state:
```

Comment:
Since a worker will only ever execute tasks serially, we can store the current task in a global state, then only need to patch stdio once and use this global state. Could be a better way to do it